genesyscloud/group/resource_genesyscloud_group.go

```diff
@@ -0,0 +1,333 @@
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+	visibility := d.Get("visibility").(string)
+	rulesVisible := d.Get("rules_visible").(bool)
+
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	gp := getGroupProxy(sdkConfig)
+
+	diagErr := util.RetryWhen(util.IsVersionMismatch, func() (*platformclientv2.APIResponse, diag.Diagnostics) {
+		// Get current group version
+		group, resp, getErr := gp.getGroupById(ctx, d.Id())
+		if getErr != nil {
+			return resp, util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to read group %s: %s", d.Id(), getErr), resp)
+
+		}
+
+		addresses, err := buildSdkGroupAddresses(d)
+		if err != nil {
+			return nil, util.BuildDiagnosticError(resourceName, fmt.Sprintf("Error while trying to buildSdkGroupAddresses for group id: %s", d.Id()), err)
+		}
+
+		log.Printf("Updating group %s", name)
+		updateGroup := &platformclientv2.Groupupdate{
+			Version:      group.Version,
+			Name:         &name,
+			Description:  &description,
+			Visibility:   &visibility,
+			RulesVisible: &rulesVisible,
+			Addresses:    addresses,
+			OwnerIds:     lists.BuildSdkStringListFromInterfaceArray(d, "owner_ids"),
+		}
+		_, resp, putErr := gp.updateGroup(ctx, d.Id(), updateGroup)
+		if putErr != nil {
+			return resp, util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to update group %s: %s", d.Id(), putErr), resp)
+		}
+
+		return resp, nil
+	})
+	if diagErr != nil {
+		return diagErr
+	}
+
+	diagErr = updateGroupMembers(ctx, d, sdkConfig)
+	if diagErr != nil {
+		return diagErr
+	}
+
+	log.Printf("Updated group %s", name)
+	return readGroup(ctx, d, meta)
+}
+
+func deleteGroup(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	gp := getGroupProxy(sdkConfig)
+
+	util.RetryWhen(util.IsVersionMismatch, func() (*platformclientv2.APIResponse, diag.Diagnostics) {
+		// Directory occasionally returns version errors on deletes if an object was updated at the same time.
+		log.Printf("Deleting group %s", name)
+		resp, err := gp.deleteGroup(ctx, d.Id())
+		if err != nil {
+			return resp, util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to delete group %s: %s", name, err), resp)
+		}
+		return nil, nil
+	})
+
+	return util.WithRetries(ctx, 60*time.Second, func() *retry.RetryError {
+		group, resp, err := gp.getGroupById(ctx, d.Id())
+		if err != nil {
+			if util.IsStatus404(resp) {
+				log.Printf("Group %s deleted", name)
+				return nil
+			}
+			return retry.NonRetryableError(fmt.Errorf("Error deleting group %s: %s", d.Id(), err))
+		}
+
+		if group.State != nil && *group.State == "deleted" {
+			log.Printf("Group %s deleted", name)
+			return nil
+		}
+
+		/*
+		  This extra delete call is being added here because of  DEVTOOLING-485.  Basically we are in a transition
```

Comment:
This was the bug fix that started this whole thing.  I am adding an extra delete, because sometimes when we delete a record directory is quietly failing the delete so if we find the record after the delete, we are going to retry